examples/arithmetics/src/language-server/main-browser.ts

```diff
@@ -26,14 +29,28 @@ const documentChangeNotification = new NotificationType<DocumentChange>('browser
 const jsonSerializer = arithmetics.serializer.JsonSerializer;
 shared.workspace.DocumentBuilder.onBuildPhase(DocumentState.Validated, documents => {
     for (const document of documents) {
-        const json = jsonSerializer.serialize(document.parseResult.value, {
-            sourceText: true,
-            textRegions: true,
-        });
+        const json = [];
+        const module = document.parseResult.value as Module;
+        // create a json object with the all evaluations
+        if(document.diagnostics === undefined  || document.diagnostics.filter((i) => i.severity === 1).length === 0) {
+            for (const [evaluation, value] of interpretEvaluations(module)) {
+                const cstNode = evaluation.expression.$cstNode;
+                if (cstNode) {
+                    json.push({range: evaluation.expression.$cstNode?.range, text: evaluation.expression.$cstNode?.text, value: value});
+                }
+            }
+        }
+
+        // add the evaluations to the ast object
+        (module as unknown as { evaluations: Array<{
+            range: Range | undefined;
+            text: string | undefined;
+            value: number;
+        }> }).evaluations = json;
```

Comment:
One remark on this: I encourage you to change the field `evaluations` to `$evaluations` to emphasize that those data do not belong to the AST.

Otherwise I'm fine. 😀

Reply:
Aah, yes, because it's metadata